manager/manager.go

```diff
@@ -66,8 +66,11 @@ type Manager interface {
 	// Gets spec for a container.
 	GetContainerSpec(containerName string) (v2.ContainerSpec, error)
 
-	// Get derived stats for a container.
-	GetContainerDerivedStats(containerName string) (v2.DerivedStats, error)
+	// Get data for all requested containers based on the request options.
+	GetRequestedContainers(containerName string, options v2.RequestOptions) (map[string]*containerData, error)
```

Comment:
containerData wasn't meant to be publicly exposed. If we do this route we should make it safe to do so (doesn't have to be in this PR). I'm just worried about its use in Kubelet :)

We can also replace most of the stuff here with a call to this and then getting some data.